pages/overview/[id].tsx

```diff
@@ -0,0 +1,221 @@
+import useEthersProvider from '../../hooks/useEthersProvider';
+import isLSP7orLSP8 from '../../utils/isLSP7orLSP8';
+import fetchLSP4Metadata from '../../utils/fetchLSP4Metadata';
+import fetchLSP7Balance from '../../utils/fetchLSP7Balance';
+import fetchLSP8Metadata from '../../utils/fetchLSP8Metadata';
+import fetchLSP8TokensIds from '../../utils/fetchLSP8TokensIds';
+import isUP from '../../utils/isUP';
+import useWeb3Provider from '../../hooks/useWeb3Provider';
+import { Lsp7AssetType, Lsp8AssetType } from '../../contexts/AssetsContext';
+import LSP7Table from '../../components/overview/LSP7sTable';
+import LSP8Table from '../../components/overview/LSP8Table';
+import { luksoImg } from '../../constants';
+import { LSP4Metadata } from '../../interfaces/lsps';
+
+const AdressOverview: NextPage = () => {
+  const router = useRouter();
+  const [addressError, setAddressError] = useState<string>('');
+  const [isUniversalProfile, setIsUniversalProfile] = useState<boolean>(false);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const [showLSP7, setShowLSP7] = useState<boolean>(true);
+
+  const { lsp7Assets, setLsp7Assets, lsp8Assets, setLsp8Assets } =
+    useContext(AssetsContext);
+  const provider = useEthersProvider() as ethers.providers.BaseProvider;
+
+  //ERC725 does not support ethers provider
+  const web3Provider = useWeb3Provider();
+
+  const createLSP8Object = (
+    NFTLSP4MetadataJSON: LSP4Metadata,
+    tokenId: string,
+    collectionName: string,
+    collectionSymbol: string,
+    assetAddress: string,
+    collectionLSP4Metadata: LSP4Metadata,
+  ): Lsp8AssetType => {
+    const { description, links, images, assets, icons } =
+      NFTLSP4MetadataJSON.LSP4Metadata;
+
+    const lsp8AssetObject = {
+      tokenId,
+      description,
+      image: images[0][0]?.url ? images[0][0].url : luksoImg,
+      icon: icons[0]?.url ? icons[0].url : luksoImg,
+      collectionName,
+      collectionSymbol,
+      collectionAddress: assetAddress,
+      collectionDescription: collectionLSP4Metadata.LSP4Metadata.description,
+      collectionImage: collectionLSP4Metadata.LSP4Metadata.images[0][0]?.url
+        ? collectionLSP4Metadata.LSP4Metadata.images[0][0]?.url
+        : luksoImg,
+      collectionIcon: collectionLSP4Metadata.LSP4Metadata.icons[0]?.url
+        ? collectionLSP4Metadata.LSP4Metadata.icons[0]?.url
+        : luksoImg,
+    };
+    return lsp8AssetObject as Lsp8AssetType;
+  };
+
+  const getAssets = async (address: string) => {
+    //fetch all received assets for specific up address
+    const receivedAssets = await fetchReceivedAssets(address, web3Provider);
+    //loop over assets
+    for await (const assetAddress of receivedAssets) {
+      //figure out if asset is LSP7 or LSP8
+      const assetType = await isLSP7orLSP8(assetAddress, provider);
+      //if LSP7 then fetch LSP7 assets
+      if (assetType === 'LSP7') {
+        const LSP4MetadataResponse = await fetchLSP4Metadata(
```

Comment:
I think you can use `Promise.all()` here.
You map through each elements, prepare your promises, then you Promise.all.Might save load time.